ompi/mca/pml/ob1/pml_ob1_rdma.c

```diff
@@ -124,8 +124,8 @@ size_t mca_pml_ob1_rdma_pipeline_btls_count (mca_bml_base_endpoint_t* bml_endpoi
         /* NTH: go ahead and use an rdma btl if is the only one */
         bool ignore = !mca_pml_ob1.use_all_rdma;
 
-        for (int i = 0 ; i < num_eager_btls && ignore ; ++i) {
-            mca_bml_base_btl_t *eager_btl = mca_bml_base_btl_array_get_index (&bml_endpoint->btl_eager, i);
+        for (int j = 0 ; j < num_eager_btls && ignore ; ++j) {
+            mca_bml_base_btl_t *eager_btl = mca_bml_base_btl_array_get_index (&bml_endpoint->btl_eager, j);
```

Comment:
@hjelmn should this be an i or j passed into `mca_bml_base_btl_array_get_index()` - I took a guess at `j`

Reply:
Didn't take long to find a bug being hidden by shadowing. Yes. Both should be j.

Reply:
There was no bug here, the previous code was correctrly using the inner i.

Reply:
The problem is i was used both in the inner and outer loop.

Reply:
unclear yes, incorrect no because the compiler would have used the inner loop index variable in the mca_bml_base_btl_array_get_index call.

Reply:
Perhaps we should distinguish from "technically correct" and "good coding practice" here. While it may well be true that the code is technically correct as written, it is not good coding practice since it results in unclear code that could easily evolve into a true bug at some later time. Removing the shadowing does no harm and results in clearer, less error prone code...and thus, is a good thing to do.